sympy/polys/ring_series.py

```diff
@@ -490,9 +622,344 @@ def rs_exp(p, x, prec):
     r = rs_series_from_list(p, c, x, prec)
     return r
 
+def _atan_series(p, iv, prec):
+    ring = p.ring
+    mo = ring(-1)
+    c = [-mo]
+    p2 = rs_square(p, iv, prec)
+    for k in range(1, prec):
+        c.append(mo**k/(2*k + 1))
+    s = rs_series_from_list(p2, c, iv, prec)
+    s = rs_mul(s, p, iv, prec)
+    return s
+
+def rs_atan(p, x, prec):
+    """
+    The arctangent of a series
+
+    Returns the series expansion of the atan of p, about 0.
+
+    Examples
+    ========
+
+    >>> from sympy.polys.domains import QQ
+    >>> from sympy.polys.rings import ring
+    >>> from sympy.polys.ring_series import rs_atan
+    >>> R, x = ring('x', QQ)
+    >>> rs_atan(x, x, 8)
+    -1/7*x**7 + 1/5*x**5 - 1/3*x**3 + x
+
+    See Also
+    ========
+
+    atan
+    """
+    if _has_constant_term(p, x):
+        raise NotImplementedError('Polynomial must not have constant term in \
+              the series variables')
+    ring = p.ring
+
+    # Instead of using a closed form formula, we differentiate atan(p) to get
+    # `1/(1+p**2) * dp`, whose series expansion is much easier to calculate.
+    # Finally we integrate to get back atan
+    if x in ring.gens:
+        dp = p.diff(x)
+        p1 = rs_square(p, x, prec) + ring(1)
+        p1 = rs_series_inversion(p1, x, prec - 1)
+        p1 = rs_mul(dp, p1, x, prec - 1)
+        return rs_integrate(p1, x)
+    else:
+        return _atan_series(p, x, prec)
+
+def _tan1(p, x, prec):
+    """
+    Helper function of ``rs_tan``
```

Comment:
I saw you imported this in one of the doctests above. What exactly is this function doing? You should improve the docstring.